internal/auth/client.go

```diff
@@ -0,0 +1,138 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// Client provides an HTTP client with automatic token management.
+// Thread-safe for concurrent use by multiple goroutines/services.
+// The mutex (mu) protects credentials during refresh operations when
+// multiple API calls might trigger refresh simultaneously.
+type Client struct {
+	httpClient  *http.Client
+	provider    *Provider
+	clientID    string
+	credentials *Credentials
+	mu          sync.RWMutex // Protects credentials for concurrent access
```

Comment:
As discussed in slack. The mutex here doesn't feel right, I think logically everything is fine, but I wonder if pushing this down to the `Credentials` type itself may make more sense.

To be clear, I'm not saying make any changes now, more of a "lets think if there is a cleaner way to do this".

Reply:
I think it's fine the way it is.